homeassistant/components/edilkamin/entity.py

```diff
@@ -0,0 +1,80 @@
+"""Edilkamin integration entity."""
+from typing import Any, cast
+
+import edilkamin
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+
+from .const import LOGGER
+
+power_to_hvac = {
+    edilkamin.Power.OFF: HVACMode.OFF,
+    edilkamin.Power.ON: HVACMode.HEAT,
+}
+hvac_mode_to_power = {hvac: power for (power, hvac) in power_to_hvac.items()}
+
+
+class EdilkaminEntity(ClimateEntity):
+    """Representation of a stove."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+
+    def __init__(
+        self,
+        username: str,
+        password: str,
+        mac_address: str,
+    ) -> None:
+        """Create the Edilkamin entity.
+
+        Use:
+        - the username/password to login
+        - the MAC address that identifies the stove
+        """
+        self._attr_name = f"Stove ({mac_address})"
+        self._attr_unique_id = mac_address
+        self._username = username
+        self._password = password
+        self._mac_address = mac_address
+        self._device_info = None
+
+    def refresh_token(self) -> str:
+        """Login to refresh the token."""
+        return cast(str, edilkamin.sign_in(self._username, self._password))
+
+    def update(self) -> None:
```

Comment:
I strongly suggest using a [`DataUpdateCoordinator`](https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities) instead